src/oracles.jl

```diff
@@ -303,3 +303,69 @@ function compute_extreme_point(
     end
     return storage
 end
+
+"""
+    ChasingGradientLMO{LMO,T}
+
+Oracle wrapping another one of type LMO to boost FW with gradient pursuit.
+Pursuit rounds end once the alignment improvement with the direction of the gradient 
+is lower than `improv_tol` or `max_rounds_number` is reached.
+See the [paper](https://arxiv.org/abs/2003.06369)
+
+All keyword arguments are passed to the inner LMO.
+"""
+struct ChasingGradientLMO{LMO<:LinearMinimizationOracle,T} <: LinearMinimizationOracle
+    inner::LMO
+    max_rounds_number::Int
+    improv_tol::T
+end
+
+function compute_extreme_point(lmo::ChasingGradientLMO, direction; x, kwargs...)
+    d = 0 * direction
+    Λ = 0.0
+    if norm(direction) <= eps(eltype(direction))
+        return compute_extreme_point(lmo.inner, direction, x=x, kwargs...)
+    end
+    flag = true
+    function align(d1, d2)
+        if norm(d1) <= eps(eltype(d1)) || norm(d2) <= eps(eltype(d2))
+            return -1.0
+        else
+            return fast_dot(d1, d2) / (norm(d1) * norm(d2))
```

Comment:
this will be type-unstable if the eltype of d1 and d2 are not Float64, because one branch of the condition will return one type and the other Float64